api/grpcserver.go

```diff
@@ -394,21 +394,36 @@ func (svr *GRPCServer) GetLogs(ctx context.Context, in *iotexapi.GetLogsRequest)
 		return nil, status.Error(codes.InvalidArgument, "empty filter")
 	}
 	var (
-		ret []*iotextypes.Log
-		err error
+		logs   []*action.Log
+		hashes []hash.Hash256
+		err    error
 	)
 	switch {
 	case in.GetByBlock() != nil:
-		ret, err = svr.coreService.LogsInBlockByHash(logfilter.NewLogFilter(in.GetFilter(), nil, nil), hash.BytesToHash256(in.GetByBlock().BlockHash))
+		blkHash := hash.BytesToHash256(in.GetByBlock().BlockHash)
+		logs, err = svr.coreService.LogsInBlockByHash(logfilter.NewLogFilter(in.GetFilter()), blkHash)
+		if err != nil {
+			break
+		}
+		hashes = make([]hash.Hash256, 0, len(logs))
+		for range logs {
+			hashes = append(hashes, blkHash)
+		}
```

Comment:
why not compose `retval` in this for loop directly?

Reply:
To be solved in another pr